raml-parser-2/src/main/java/org/raml/v2/internal/impl/v10/phase/LibraryLinkingTransformation.java

```diff
@@ -28,10 +28,17 @@
 
 import java.io.IOException;
 import java.io.InputStream;
+import java.util.HashSet;
+import java.util.Set;
 
 public class LibraryLinkingTransformation implements Transformer
 {
 
+    private static final ThreadLocal<Set<String>> absoluteLocations = new ThreadLocal<>();
+    static {
+        absoluteLocations.set(new HashSet<String>());
```

Comment:
No problem (I agree). However, the injected builder will have to maintain a history of opened files.  Which is correct.